parquet/src/main/java/org/apache/iceberg/parquet/ParquetUtil.java

```diff
@@ -147,9 +158,39 @@ public static Metrics footerMetrics(ParquetMetadata metadata, Stream<FieldMetric
       upperBounds.remove(fieldId);
     }
 
+    updateFromFieldMetrics(fieldMetricsMap, metricsConfig, fileSchema, lowerBounds, upperBounds);
+
     return new Metrics(rowCount, columnSizes, valueCounts, nullValueCounts,
-        MetricsUtil.createNanValueCounts(fieldMetrics, metricsConfig, fileSchema),
-        toBufferMap(fileSchema, lowerBounds), toBufferMap(fileSchema, upperBounds));
+        MetricsUtil.createNanValueCounts(fieldMetricsMap.values().stream(), metricsConfig, fileSchema),
+        toBufferMap(fileSchema, lowerBounds),
+        toBufferMap(fileSchema, upperBounds));
+  }
+
+  private static void updateFromFieldMetrics(
+      Map<Integer, FieldMetrics> idToFieldMetricsMap, MetricsConfig metricsConfig, Schema schema,
+      Map<Integer, Literal<?>> lowerBounds, Map<Integer, Literal<?>> upperBounds) {
+    idToFieldMetricsMap.entrySet().forEach(entry -> {
+      int fieldId = entry.getKey();
+      FieldMetrics metrics = entry.getValue();
+      MetricsMode metricsMode = MetricsUtil.metricsMode(schema, metricsConfig, fieldId);
+
+      // only check for MetricsModes.None, since we don't truncate float/double values.
+      if (metricsMode != MetricsModes.None.get()) {
+        if (metrics.upperBound() == null) {
+          // upper and lower bounds will both null or neither
+          lowerBounds.remove(fieldId);
+          upperBounds.remove(fieldId);
+        } else if (metrics.upperBound() instanceof Float) {
```

Comment:
I think this should be `metrics instanceof FloatFieldMetrics` instead of checking the bound.

Reply:
Actually due to having to modify the option writer to return correct value count/null value count, this can go different directions:
- we either let the option writer to explicitly check for float/double writer and create the corresponding field metric object type (also have to make float/double field metrics' constructor public)
- or we create the generic field metric type in option writer, and continue to check instanceof on bounds here

For now I decided latter for better flexibility, but don't have strong opinion either way. Please let me know your thoughts!

Reply:
Oh, I see. Good point. Let's go with this for now. In the future, we could keep the value class in the metrics, so you could do something like this:

```java
  if (metrics.javaType() instanceof Float) { ... }
```

But that's probably not worth it at this point.